gateway/src/context.rs

```diff
@@ -20,37 +21,20 @@ impl ServerContext {
         switch_config: SwitchConfig,
         log: &Logger,
     ) -> Result<Arc<Self>, StartupError> {
-        let mgmt_switch = Arc::new(
-            ManagementSwitch::new(
-                switch_config,
-                &Arc::new(TempNoopHostPhase2RecoveryProvider),
-                log,
-            )
-            .await?,
-        );
+        // The capacity here is the maximum number of recovery images we're
+        // willing to keep in memory - we only ever expect to be serving one
+        // (whatever the most recent one is).
+        let host_phase2_provider =
+            Arc::new(InMemoryHostPhase2Provider::with_capacity(1));
```

Comment:
Maybe this `1` should be from the MGS config file instead? Happy to make that change if you think so.

Reply:
Moved to config in 44443b48